src/funcx_common/task_storage/base.py

```diff
@@ -32,3 +32,19 @@ def get_result(self, task: TaskProtocol) -> t.Optional[str]:
         Returns a string if a result was found, and None otherwise.
         Raises a storage exception if retrieval failed
         """
+
+    @abc.abstractmethod
+    def store_payload(self, task: TaskProtocol, payload: str) -> None:
```

Comment:
Are payloads a string?

The web service has been using `json.dumps`/`json.loads` on them since before `funcx-common` existed:
https://github.com/funcx-faas/funcx-web-service/blob/d82b3f340832cc0844287f173ccee55016049f64/funcx_web_service/models/tasks.py#L109
and I preserved this with the `RedisField` rewrite as you can see in the current version:
https://github.com/funcx-faas/funcx-web-service/blob/05067008328c4f73337379f9bca3ae8482d16681/funcx_web_service/models/tasks.py#L62

If payloads are actually strings, not objects, then we have some sort of problem to work out here. Best case, it's just an unnecessary quoting and escaping pass we can remove, but I'm somewhat worried about trying to do work with the `payload` without fixing that.

I'll proceed with the rest of my review on the assumption that `payload` is a string and that we're going to fix the various task implementations to stop doing `json.{dumps,loads}`

Reply:
I believe so. I don't think the json loading is necessary. It will take a bit of testing to figure this out, but in order to do that, we need to get these changes out. Maybe even an alpha release will be sufficient to get the next steps unblocked.